src/main/src/main/java/org/geoserver/config/impl/GeoServerImpl.java

```diff
@@ -448,8 +453,22 @@ public void reload() throws Exception {
             // reload configuration
             GeoServerLoaderProxy loader = GeoServerExtensions.bean(GeoServerLoaderProxy.class);
             synchronized (org.geoserver.config.GeoServer.CONFIGURATION_LOCK) {
-                getCatalog().getResourcePool().dispose();
-                loader.reload();
+                getCatalog().getResourcePool().dispose();                
+
+                if(catalog != null) {
+                    dispose();
+                    
+                    //reload catalog, make sure we reload the underlying catalog, not any wrappers
+                    Catalog oldCatalog = getCatalog();
```

Comment:
This is actually "the" catalog, the one that we are getting as a parameter is the one that we will copy from, and that will be thrown away right? If so, better name them newCatalog (the parameter) and catalog (the actual gs catalog)